presentation/src/main/java/com/mashup/gabbangzip/sharedalbum/presentation/ui/main/MainViewModel.kt

```diff
@@ -0,0 +1,31 @@
+package com.mashup.gabbangzip.sharedalbum.presentation.ui.main
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.google.android.gms.tasks.OnCompleteListener
+import com.google.firebase.messaging.FirebaseMessaging
+import com.mashup.gabbangzip.sharedalbum.domain.usecase.firebase.RegisterFcmTokenUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class MainViewModel @Inject constructor(
+    private val registerFcmTokenUseCase: RegisterFcmTokenUseCase,
+) : ViewModel() {
+
+    fun registerFcmToken() {
+        FirebaseMessaging.getInstance().token.addOnCompleteListener(
+            OnCompleteListener { task ->
+                if (!task.isSuccessful) {
+                    return@OnCompleteListener
+                }
+
+                viewModelScope.launch {
+                    val token = task.result
+                    registerFcmTokenUseCase(token)
+                }
+            },
+        )
```

Comment:
```suggestion
        FirebaseMessaging.getInstance().token.addOnCompleteListener { task ->
            if (!task.isSuccessful) {
                return@addOnCompleteListener
            }

            viewModelScope.launch {
                val token = task.result
                registerFcmTokenUseCase(token)
            }
        }
```
엄청 사소하긴 한데..!! OnCompleteListener 객체 만들지 않고 바로 람다함수 쓰면 더욱 깔끔하게 코드를 작성할 수 있을 것 같아!

Reply:
와 그냥 코드 파이어베이스 사이트에서 긁어 쓰다보니 생각을 못했네..! 바로 수정할게 고마워~ 하나도 안사소해. 이거 큰일이야.